scripts/common.js

```diff
@@ -0,0 +1,102 @@
+const projects = [
+  {
+    name: "apisix-ingress-controller",
+    branch: "master",
+    hasChangelog: true,
+    changelogExtractor: function (changelogs, targetRelease) {
+      return changelogExtractor(changelogs, targetRelease, "#", true);
+    }
+  },
+  {
+    name: "apisix",
+    branch: "master",
+    hasChangelog: true,
+    changelogExtractor: function (changelogs, targetRelease) {
+      return changelogExtractor(changelogs, targetRelease, "##", false);
+    }
+  },
+  {
+    name: "apisix-dashboard",
+    branch: "master",
+    hasChangelog: true,
+    changelogExtractor: function (changelogs, targetRelease) {
+      return changelogExtractor(changelogs, targetRelease, "#", false);
+    }
+  },
+  {
+    name: "apisix-docker",
+    branch: "master",
+    hasChangelog: false,
+  },
+  {
+    name: "apisix-helm-chart",
+    branch: "master",
+    hasChangelog: false,
+  },
+  {
+    name: "apisix-java-plugin-runner",
+    branch: "main",
+    hasChangelog: true,
+    changelogExtractor: function (changelogs, targetRelease) {
+      return changelogExtractor(changelogs, targetRelease, "##", true);
+    }
+  },
+  {
+    name: "apisix-go-plugin-runner",
+    branch: "master",
+    hasChangelog: true,
+    changelogExtractor: function (changelogs, targetRelease) {
+      return changelogExtractor(changelogs, targetRelease, "##", true);
+    }
+  }
+];
+
+const languages = ["en", "zh", "es"];
+
+/**
+ * changelog extractor for each version
+ * @param {string} changelogs CHANGELOG.md contents
+ * @param {Object} targetRelease result from GitHub API
+ * @param {string} versionStartsWith version line markdown symbol
+ * @param {boolean} reserveVersionEnds0 reserve the `.0` on version end (e.g. `.0` of `1.5.0`)
+ */
```

Comment:
Thumbs up to code comments

Reply:
Haha, and is it accurate? (If I don't make records, I will soon forget the meaning of the parameters)